### ISSUE TITLE
`use` animation accessibility concerns

### ISSUE THREAD

**Issue by synecdokey** [gleam-lang/language-tour#179]
The current animation loops indefinitely, while also ignoring `prefers-reduced-motion` preferences. This fails [Success Criterion 2.2.2 Pause, Stop, Hide](https://www.w3.org/TR/WCAG21/#pause-stop-hide).

There is [guidance available](https://www.w3.org/WAI/WCAG21/Understanding/pause-stop-hide.html) on the topic:

> **An essential animation can be paused without affecting the activity**
> A website helps users understand 'how things work' through animations that demonstrate processes. Animations have "pause" and "restart" buttons.

I don't want to be prescribing a solution here, but it seems like adding controls may be a solution to at least the accessibility failure, while there might also be a way to deliver the underlying example on `prefers-reduced-motion` (since it does a great job of having the proper markup for screenreaders already)

**Comment by lpil** [comment]
Thank you! That background seems to also be rendering incorrectly for folks due to the absolute positioning. I think we'll need to revert this and try again.

**Comment by ollie-dot-earth** [comment]
So a pause, manual toggle and remove button. And slowing the transitions for `prefers-reduced-motion` ?

And figuring out a more reliable way to size it 🤔

**Comment by ollie-dot-earth** [comment]
Im thinking the remove button would just remove the classes (and stop the periodic transitions). Which would give it the old "This turns into that" text back

**Comment by gungun974** [comment]
> Im thinking the remove button would just remove the classes (and stop the periodic transitions). Which would give it the old "This turns into that" text back

I would more imagine/see the animation to be paused by default with a clear button (like on YT video) to play the animation. What I mean I don't think the animation should autoplay when the user arrived on the page.

**Comment by ollie-dot-earth** [comment]
This would be fine for `prefers-reduced-motion` right? (Ignoring the improper codeblock styling rn)

https://github.com/user-attachments/assets/f6d38677-2c7e-4399-9163-d95459abc890

**Comment by ollie-dot-earth** [comment]
Also got this(size based on content and then transitioning it) working now

https://github.com/user-attachments/assets/0a2bf496-548d-411b-be61-74165b93ca2b

**Comment by ollie-dot-earth** [comment]
Now sizes correctly on: 
- Vivaldi (chromium) 
- Firefox 
- Gnome browser (webkit)
- Brave

I cant test on arc unfortunately

**Comment by ollie-dot-earth** [comment]
> > Im thinking the remove button would just remove the classes (and stop the periodic transitions). Which would give it the old "This turns into that" text back
> 
> I would more imagine/see the animation to be paused by default with a clear button (like on YT video) to play the animation. What I mean I don't think the animation should autoplay when the user arrived on the page.

Something like this? @gungun974 

https://github.com/user-attachments/assets/f019d770-1c85-4537-8c2d-751c9ef8460a

**Comment by ginkogruen** [comment]
I personally think a toggle button which converts the normal _callback style_ to the _`use` style_ (and vice-versa) could also make sense here as it would allow everyone to process the animation at their own speed giving them time to grasp either of the two states before toggling the animation again (as opposed to a toggle which just toggles the animation loop on or off).

**Comment by ollie-dot-earth** [comment]
Way ahead of you :) 

<img width="459" height="327" alt="Image" src="https://github.com/user-attachments/assets/175b54ef-1eee-4f80-ad5a-6bcdb71e7a62" />

**Comment by gungun974** [comment]
> > > Im thinking the remove button would just remove the classes (and stop the periodic transitions). Which would give it the old "This turns into that" text back
> > 
> > 
> > I would more imagine/see the animation to be paused by default with a clear button (like on YT video) to play the animation. What I mean I don't think the animation should autoplay when the user arrived on the page.
> 
> Something like this? [@gungun974](https://github.com/gungun974)
> 
>  Screencast.From.2026-07-18.11-18-59.webm

I was thinking about something like that but more like if it was a video player where you got a big play button and everything is a bit dark and when it’s played the button move to the bottom right of the windows to give you the power to pause it.

**Comment by ollie-dot-earth** [comment]
What do we think about buttons like this https://github.com/ollie-dot-earth/gleam-language-tour/pull/1 ?

**Comment by ginkogruen** [comment]
> What do we think about buttons like this https://github.com/ollie-dot-earth/gleam-language-tour/pull/1 ?

I like the buttons but in my opinion the box should maybe not resize in order to have the buttons stay at the same position.

**Comment by ollie-dot-earth** [comment]
> > What do we think about buttons like this [ollie-dot-earth#1](https://github.com/ollie-dot-earth/gleam-language-tour/pull/1) ?
> 
> I like the buttons but in my opinion the box should maybe not resize in order to have the buttons stay at the same position.

Any Ideas on how to improve it? We could have the buttons stay where they are but styled like that i guess?

**Comment by ginkogruen** [comment]
> Any Ideas on how to improve it? We could have the buttons stay where they are but styled like that i guess?

My main gripe is the horizontal change in size so I would say just keep the box the size it needs for the `use` state so the buttons don't move. 

With the vertical size I can't decide if having an empty line would be better or if that should still change in size. Maybe an empty line there is the better option as well as otherwise I text below the box would need to adjust as well (if I'm not missing anything).

So horizontal size: `use` state, vertical size: `callback` state?

**Comment by strawmelonjuice** [comment]
> > Any Ideas on how to improve it? We could have the buttons stay where they are but styled like that i guess?
> 
> My main gripe is the horizontal change in size so I would say just keep the box the size it needs for the `use` state so the buttons don't move.
> 
> With the vertical size I can't decide if having an empty line would be better or if that should still change in size. Maybe an empty line there is the better option as well as otherwise I text below the box would need to adjust as well (if I'm not missing anything).
> 
> So horizontal size: `use` state, vertical size: `callback` state?

I did try this yes! I in the end took the resizing approach. I can see why maybe picking a set size based on screen is better. I'll ammend later!
